### ISSUE TITLE
Make some flag for case-insensitive processing

### ISSUE THREAD

**Issue by jakewilliami** [guo-yong-zhi/WordCloud.jl#3]
I think that case insensitive might be nice.  If there isn't already functionality.

P.S., I think docs would be really nice for this package.  Happy to help you do that when I have a little more time :-)

Thanks for this package!

**Comment by guo-yong-zhi** [comment]
Yes, there is a default function [`casemerge!`](https://github.com/guo-yong-zhi/WordCloud.jl/blob/da0906fffe0e9d97de34b93837654605466297e3/src/textprocessing.jl#L98) to handle the case of words.
```julia
using WordCloud
text = "Julia julia julia"
@show processtext(text) #counterprocessor=casemerge!
@show processtext(text, counterprocessor=identity)
```
And, I would really appreciate it if you can help improve the docs. It is a little bit hard for a non-native English speaker like me.

**Comment by jakewilliami** [comment]
Is there a way to use casemerge with the counter dict?

```julia
julia> using WordCloud;

julia> countmap = Dict("Julia" => 1, "julia" => 2);

julia> processtext(countmap, counterprocessor = casemerge!) # currently this throws an error
3 words
1 different word
1 legal word
1 huge word
(["Julia"], [1])
```

Or perhaps I am misunderstanding?

**Comment by guo-yong-zhi** [comment]
`countmap` need not to be processed as text, `wordcloud` can accept a `Dict` as its input, just
```julia
using WordCloud;
countmap = Dict("Julia" => 1, "julia" => 2)
wordcloud(casemerge!(countmap))
```

**Comment by jakewilliami** [comment]
Ah, my bad.  I was using processtext because I wanted to use minfrequency but I couldn’t find out how to use minfrequency with just wordcloud.

**Comment by guo-yong-zhi** [comment]
Oh, yes, you are right. In some case, we may want the `processtext` to process a weight Dict. The interface needs to be improved. Currently you can call the function this way:
```julia
processtext(casemerge!(countmap))
```

**Comment by jakewilliami** [comment]
Ahh I see, thank you.  I will let you know when I get around to doing the documentation.  Hopefully I can get some time some time this weekend.

**Comment by guo-yong-zhi** [comment]
Thank you. I'm looking forward to it.

**Comment by guo-yong-zhi** [comment]
Now, after 8e5137b7f6345d2c81e222041762fb2b523592c7, the new usage is:
```julia
using WordCloud
text = "Word word word words" #or text = ["Word" => 1, "word" => 2, "words" => 1]
@show processtext(text) #process=casemerge!∘lemmatize!, case-insensitive
@show processtext(text, process=lemmatize!) #only lemmatize!, case-sensitive
@show processtext(text, process=casemerge!) #only casemerge!
@show processtext(text, process=identity) #neither
```

**Comment by jakewilliami** [comment]
Great!  May you please send me an email so that I can ask some questions about documentation?  My email is on my profile (or I think in my GH pages website)

**Comment by guo-yong-zhi** [comment]
The bug was fixed at v0.7.0. I closed the issue and we can discuss other things in the email.
